docs/content/advanced/backend/payment/how-to-create-payment-provider.md

```diff
@@ -53,51 +53,44 @@ For example, create the file `src/services/my-payment.ts` with the following con
 <!-- eslint-disable max-len -->
 
 ```ts title=src/services/my-payment.ts
-import { 
-  AbstractPaymentService, 
-  Cart, Data, Payment, PaymentSession, 
-  PaymentSessionStatus, TransactionBaseService, 
-} from "@medusajs/medusa"
+import { AbstractPaymentService, Context, Data, Payment, PaymentSession, PaymentSessionStatus, TransactionBaseService } from "@medusajs/medusa"
 import { EntityManager } from "typeorm"
 
 class MyPaymentService extends AbstractPaymentService<TransactionBaseService> {
   protected manager_: EntityManager
   protected transactionManager_: EntityManager
 
-  getPaymentData(paymentSession: PaymentSession): Promise<Data> {
+  async getPaymentData(paymentSession: PaymentSession): Promise<Data> {
     throw new Error("Method not implemented.")
   }
-  updatePaymentData(paymentSessionData: Data, data: Data): Promise<Data> {
+  async updatePaymentData(paymentSessionData: Data, data: Data): Promise<Data> {
     throw new Error("Method not implemented.")
   }
-  createPayment(cart: Cart): Promise<Data> {
+  async createPayment(context: Context): Promise<PaymentSessionResponse> {
     throw new Error("Method not implemented.")
   }
-  retrievePayment(paymentData: Data): Promise<Data> {
+  async retrievePayment(paymentData: Data): Promise<Data> {
     throw new Error("Method not implemented.")
   }
-  updatePayment(paymentSessionData: Data, cart: Cart): Promise<Data> {
+  async updatePayment(paymentSessionData: Data, context: Context): Promise<PaymentSessionResponse> {
     throw new Error("Method not implemented.")
   }
-  authorizePayment(
-    paymentSession: PaymentSession,
-    context: Data
-  ): Promise<{ data: Data; status: PaymentSessionStatus; }> {
+  async authorizePayment(paymentSession: PaymentSession, context: Data): Promise<{ data: Data; status: PaymentSessionStatus; }> {
     throw new Error("Method not implemented.")
   }
-  capturePayment(payment: Payment): Promise<Data> {
+  async capturePayment(payment: Payment): Promise<Data> {
```

Comment:
todo: Promise<PaymentData>

Reply:
Not sure what this comment refers to. Can you explain more?

Reply:
Sorry, Promise<PaymentData>